idx-kotlin/src/main/java/com/okta/idx/kotlin/dto/IdxRemediation.kt

```diff
@@ -63,6 +63,9 @@ class IdxRemediation internal constructor(
     /** The set of authenticators associated with this remediation. */
     val authenticators: IdxAuthenticatorCollection,
 
+    /** The IdP associated with a remediation of type [Type.REDIRECT_IDP]. */
+    val idp: Idp?,
```

Comment:
So all remediations, whether or not they're IDP, will have this property?  Would subclassing be a convenient approach here, so only those Social Auth remediation types will contain this property?

The reason I ask is that there will be other types of custom properties that only some remediations support.

Reply:
Yeah, I thought the same thing. I definitely think we need to discuss approach here. I had subclassing on my list of options as well. I don't love just tagging new things on to the generic remediation, as there will be a lot once we implement everything. Planning on having an offline discussion, and following up with a future refactor.